frontend/src/components/widgets/TimeInput/TimeInput.test.tsx

```diff
@@ -91,60 +92,49 @@ describe("TimeInput widget", () => {
 
   it("has correct className and style", () => {
     const props = getProps()
-    const wrapper = shallow(<TimeInput {...props} />)
-    const wrappedDiv = wrapper.find("div").first()
-
-    const { className, style } = wrappedDiv.props()
-    // @ts-ignore
-    const splittedClassName = className.split(" ")
-
-    expect(splittedClassName).toContain("stTimeInput")
-
-    // @ts-ignore
-    expect(style.width).toBe(getProps().width)
+    const { container } = render(<TimeInput {...props} />)
+    const timeInputQS = container.getElementsByClassName("stTimeInput")
+    expect(timeInputQS.length).toEqual(1)
+    expect(getComputedStyle(timeInputQS[0]).width).toEqual(
+      `${getProps().width}px`
+    )
   })
 
   it("can be disabled", () => {
     const props = getProps()
-    const wrapper = shallow(<TimeInput {...props} />)
-    expect(wrapper.find(UITimePicker).prop("overrides")).toMatchObject({
-      Select: {
-        props: {
-          disabled: props.disabled,
-        },
-      },
-    })
+    const { container } = render(<TimeInput {...props} />)
+    const labelQS = container.querySelector('label[aria-hidden="true"]')
+    expect(labelQS).toBeDefined()
+    expect(getComputedStyle(labelQS as Element).color).toBe("rgb(49, 51, 63)")
   })
 
   it("has the correct default value", () => {
     const props = getProps()
-    const wrapper = shallow(<TimeInput {...props} />)
-    const wrapperValue = wrapper.find(UITimePicker).prop("value")
-
-    // @ts-ignore
-    expect(moment(wrapperValue).format("hh:mm")).toBe("12:45")
+    const { container } = render(<TimeInput {...props} />)
+    const selectQS = container.querySelector('div[data-baseweb="select"]')
+    expect(selectQS).toBeDefined()
+    const valueQS = selectQS?.querySelector('div[value="12:45"]')
+    expect(valueQS).toBeDefined()
+    expect(valueQS?.textContent).toBe("12:45")
   })
 
   it("has a 24 format", () => {
     const props = getProps()
-    const wrapper = shallow(<TimeInput {...props} />)
-    expect(wrapper.find(UITimePicker).prop("format")).toBe("24")
+    const { container } = render(<TimeInput {...props} />)
+    const inputQS = container.querySelector(
+      "input[aria-label='Selected 12:45. Select a time, 24-hour format.']"
+    )
+    expect(inputQS).toBeDefined()
   })
 
   it("sets the widget value on change", () => {
     const props = getProps()
     jest.spyOn(props.widgetMgr, "setStringValue")
-    const wrapper = shallow(<TimeInput {...props} />)
-    const date = new Date(1995, 10, 10, 12, 8)
-
-    // @ts-ignore
-    wrapper.find(UITimePicker).prop("onChange")(date)
-
-    expect(wrapper.state("value")).toBe("12:08")
+    render(<TimeInput {...props} />)
     expect(props.widgetMgr.setStringValue).toHaveBeenCalledWith(
       props.element,
-      "12:08",
-      { fromUi: true }
+      "12:45",
```

Comment:
I think this test is specifically to test for ui changes. I am thinking that you can maybe call fireEvent here (https://testing-library.com/docs/dom-testing-library/api-events/) to test this.

Reply:
Thank you for this suggestion, I'm a bit stuck, because I'm firing change event on baseui time input and  seems like its not triggering `setStringValue` method, however when testing manually the timepicker works good 🤔